aws_protocol_adaptor/device_client/aws_nvmsgbroker.c

```diff
@@ -243,6 +243,44 @@ NvDsMsgApiErrorType nvds_msgapi_send_async(NvDsMsgApiHandle h_ptr, char *topic,
 	return NVDS_MSGAPI_OK;
 }
 
+/* ************************************************************************* */
+// Subscribe def
+/* ************************************************************************* */
+
+void subscribe_callback_handler(AWS_IoT_Client *pClient, char *topicName, uint16_t topicNameLen,
+								IoT_Publish_Message_Params *params, void *pData)
+{
+	IOT_INFO("Subscribe callback");
+	IOT_INFO("%.*s\t%.*s", topicNameLen, topicName, (int)params->payloadLen, (char *)params->payload);
+}
+
+NvDsMsgApiErrorType nvds_msgapi_subscribe(NvDsMsgApiHandle h_ptr, char **topics, int num_topics, nvds_msgapi_subscribe_request_cb_t cb, void *user_ctx)
+{
+	printf("Subscribe called\n");
+
+	if ((h_ptr == NULL) || (topics == NULL) || (num_topics <= 0))
+	{
+		IOT_ERROR("Essensial args missing for function nvds_msgapi_subscribe: %d, %d, %d\n", (h_ptr == NULL), (topics == NULL), (num_topics == NULL);
```

Comment:
Typo: "essensial"

Reply:
Thats the typo in the whole file I will correct that in all other places as well